src/librustc_resolve/diagnostics.rs

```diff
@@ -10,8 +10,111 @@
 
 #![allow(non_snake_case)]
 
+// Error messages for EXXXX errors.
+// Each message should start and end with a new line, and be wrapped to 80 characters.
+// In vim you can `:set tw=80` and use `gq` to wrap paragraphs. Use `:set tw=0` to disable.
+register_long_diagnostics! {
+
+E0154: r##"
+Imports (`use` statements) are not allowed after non-item statements, such as
+variable declarations and expression statements.
+
+Wrong example:
```

Comment:
I love this error message, but I think the "Wrong example" wording is a bit awkward (despite its incidence elsewhere). In this case you could maybe say "Here's an example that demonstrates the error:".